src/server/services/buurt/helpers.test.ts

```diff
@@ -761,4 +765,168 @@ describe('Buurt helpers', () => {
 
     expect(result).toStrictEqual(transformedResponse);
   });
+
+  describe('flatten Tests', () => {
+    test('Flattens one array', () => {
+      const positions: LatLngTuple = [4.926979845934051, 52.35619987080679];
+      const input: LatLngTuple[] = [positions];
+
+      const result = helpersForTesting.flatten(input);
+      expect(result).toStrictEqual(input);
+    });
+
+    test('Already flattened input is unchanged', () => {
+      const positions: LatLngTuple = [4.926979845934051, 52.35619987080679];
+      const input: LatLngTuple[] = [positions, positions, positions, positions];
+
+      const result = helpersForTesting.flatten(input);
+      expect(result).toStrictEqual(input);
+    });
+
+    test('Flattens multiple arrays', () => {
+      const positions: LatLngTuple = [4.926979845934051, 52.35619987080679];
+      const input: LatLngTuple[][] = [
+        [positions, positions],
+        [positions, positions],
+      ];
+
+      const result = helpersForTesting.flatten(input);
+      expect(result).toStrictEqual([
+        positions,
+        positions,
+        positions,
+        positions,
+      ]);
+    });
+
+    test('Flattens empty array', () => {
+      const input: LatLngTuple[] = [];
+      const result = helpersForTesting.flatten(input);
+      expect(result).toStrictEqual(input);
+    });
+  });
+
+  describe('filterFeaturesinRadius Tests', () => {
+    test('Keeps all feature objects that are inside the given radius', () => {
+      const coordinate = [52.36764560318806, 4.90016547381895];
+      const feature: MaFeature = {
+        type: 'Feature',
+        geometry: {
+          type: 'Polygon',
+          coordinates: [[coordinate]],
+        },
+        properties: {
+          id: '001ff443-584f-422c-b6a0-d8aac59c64d1',
+          datasetId: 'wior',
+          color: '#FEC813',
+          zIndex: datasetsForTesting.zIndexPane.WIOR,
+          datumStartUitvoering: 'Lopende werkzaamheden',
+          duur: 'Meerdaags',
+        },
+      };
+
+      const location = {
+        address: 'Amstel 1',
+        lat: 52.36764560318806,
+        lng: 4.90016547381895,
+      };
+      const features = [feature, feature, feature];
+      const radiusKilometer = 1;
+
+      const result = helpersForTesting.filterFeaturesinRadius(
+        location,
+        features,
+        radiusKilometer
+      );
+
+      expect(result.length).toBe(3);
+    });
+
+    test('Filters out all features outside of the given radius', () => {
```

Comment:
I'd say this test would be sufficient for both above and this test, no need to over-do it.

Reply:
Ik ben het er mee eens. Ik heb wel even een paar tests toegevoegd voor een paar nieuwe cases.